src/meta/src/barrier/mod.rs

```diff
@@ -210,32 +243,35 @@ pub struct GlobalBarrierManager {
 
 /// Controls the concurrent execution of commands.
 struct CheckpointControl {
+    state: BarrierWorkerState,
+
     /// Save the state and message of barrier in order.
     /// Key is the `prev_epoch`.
     command_ctx_queue: BTreeMap<u64, EpochNode>,
+    /// The barrier that are completing.
+    /// Some((`prev_epoch`, `should_pause_inject_barrier`))
+    completing_barrier: Option<(u64, bool)>,
 
     creating_streaming_job_controls: HashMap<TableId, CreatingStreamingJobControl>,
 
-    /// Command that has been collected but is still completing.
-    /// The join handle of the completing future is stored.
-    completing_task: CompletingTask,
-
     hummock_version_stats: HummockVersionStats,
 
     create_mview_tracker: CreateMviewProgressTracker,
 
-    context: GlobalBarrierManagerContext,
+    context: GlobalBarrierWorkerContext,
 }
 
 impl CheckpointControl {
```

Comment:
I think `mod.rs` is becoming a bit too large, which also affects reability. `impl CheckpointControl`, `impl GlobalBarrierWorker` and `impl GlobalBarrierManager` are somehow mixed together. I can imagine for people who are not familiar with logics here can easily get lost. Now we have clearer cut of responsibility for `CheckpointControl` and `GlobalBarrierWorker`, I am thinking maybe we should have separate files for them and maybe for other structs as well. We can do it in a separate PR or along with the future refactoring PR.

Reply:
Will reorganize the code after implementing the database isolation feature, because there are several pending PR, and if we reorganize the code in this PR, the pending PRs will have many conflicts.